ipaserver/plugins/baseuser.py

```diff
@@ -123,6 +123,24 @@ def fix_addressbook_permission_bindrule(name, template, is_new,
         template['ipapermbindruletype'] = 'anonymous'
 
 
+def update_samba_attrs(ldap, dn, entry_attrs, options):
+    samba_attrs = {'ipantlogonscript', 'ipantprofilepath',
+                   'ipanthomedirectory', 'ipanthomedirectorydrive'}
+
+    for attr in samba_attrs:
+        if options.get(attr, None):
+            if attr in entry_attrs['objectclass']:
+                entry = ldap.get_entry(dn)
+                entry[attr] = [options[attr]]
+                ldap.update_entry(entry)
+            else:
+                raise errors.ValidationError(
+                    name=attr,
+                    error=_(
+                        'can not update entry'
```

Comment:
I'd have something more reasonable here: 'Object class ipaNTUserAttrs is missing, entry cannot have Samba attributes.'

Reply:
Yes thats better i will definitelly update it.